src/Features/Core/Portable/Snippets/SnippetChange.cs

```diff
@@ -18,29 +18,29 @@ namespace Microsoft.CodeAnalysis.Snippets;
     /// </summary>
     public readonly ImmutableArray<TextChange> TextChanges;
 
-    /// <summary>
-    /// The position that the cursor should end up on
-    /// </summary>
-    public readonly int CursorPosition;
-
     /// <summary>
     /// The items that we will want to rename as well as the ordering
     /// in which to visit those items.
     /// </summary>
     public readonly ImmutableArray<SnippetPlaceholder> Placeholders;
 
+    /// <summary>
+    /// The position that the cursor should end up on
+    /// </summary>
+    public readonly int FinalCaretPosition;
+
     public SnippetChange(
         ImmutableArray<TextChange> textChanges,
-        int cursorPosition,
-        ImmutableArray<SnippetPlaceholder> placeholders)
+        ImmutableArray<SnippetPlaceholder> placeholders,
+        int finalCaretPosition)
```

Comment:
Reordered for clarity. We first insert _text changes_, then let user loop through _placeholders_ and only then move user's caret to its _final position_